File: Python/SortingAlgorithms/select_sort.py
from typing import TypeVar, List

T = TypeVar("T")

def select_sort(arr: List[T]) -> None:
    n = len(arr)
    for i in range(n - 1):
        minEl, minIdx = arr[i], i
        for j in range(i + 1, n):
            if arr[i] > arr[j]:
                minEl, minIdx = arr[j], j
        arr[minIdx],  arr[i] = arr[i], minEl

File: C++/SortingAlgorithms/SelectSort.cpp
#include "SelectSort.h"

template<typename T>
void select_sort(std::vector<T>& arr)
{
	int size = arr.size();

	for (int i = 0; i < size - 1; i++) {

		T min = arr[i];
		int minIdx = i;

		for (int j = i + 1; j < size; j++) {

			if (arr[j] < min) {
				min = arr[j];
				minIdx = j;
			}
		}

		if (minIdx == i) continue;
		
		arr[minIdx] = arr[i];
		arr[i] = min;
	}
}

template void select_sort(std::vector<int>&);